src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -188,19 +188,16 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                             {
                                 var localSymbol = MakeLocal(decl.Declaration, vdecl, kind, enclosingBinder);
                                 locals.Add(localSymbol);
-
-                                var value = vdecl.Initializer?.Value;
-                                if (value != null)
-                                {
-                                    ExpressionVariableFinder.FindExpressionVariables(this, locals, value, enclosingBinder); 
-                                }
+                                ExpressionVariableFinder.FindExpressionVariables(this, locals, vdecl.Initializer?.Value, enclosingBinder); 
                             }
                         }
                         break;
 
                     case SyntaxKind.ExpressionStatement:
                     case SyntaxKind.IfStatement:
                     case SyntaxKind.YieldReturnStatement:
+                    case SyntaxKind.ReturnStatement:
+                    case SyntaxKind.ThrowStatement:
                         ExpressionVariableFinder.FindExpressionVariables(this, locals, innerStatement, enclosingBinder.GetBinder(innerStatement) ?? enclosingBinder);
```

Comment:
>enclosingBinder.GetBinder(innerStatement) [](start = 103, length = 41)

Under what conditions would this return a non-null result, and under which conditions a null result? I'm guessing it has to do with whether or not the `SyntaxList<StatementSyntax> statements` passed in comes from the real tree or not, but some comment around here would be helpful, please.